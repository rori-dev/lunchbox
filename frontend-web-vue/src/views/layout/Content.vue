<template>
  <main class="w-full h-full">
    <router-view v-slot="{ Component }">
      <transition
        mode="out-in"
        leave-active-class="transition-opacity duration-150"
        leave-to-class="opacity-0"
        enter-active-class="transition-opacity duration-150"
        enter-from-class="opacity-0"
      >
        <component :is="Component" />
      </transition>
    </router-view>
  </main>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
export default defineComponent({ name: 'Content' })
</script>
