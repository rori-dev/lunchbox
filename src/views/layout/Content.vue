<template>
  <b-container fluid class="content">
    <router-view />
  </b-container>
</template>
