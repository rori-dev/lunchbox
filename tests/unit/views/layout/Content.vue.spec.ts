import Content from '@/views/layout/Content.vue'
import { mountUnit } from '@tests/unit/test-util'

describe('Content', () => {
  test('renders snapshot', () => {
    const wrapper = mountUnit(Content)
    expect(wrapper.element).toMatchSnapshot()
  })
})
