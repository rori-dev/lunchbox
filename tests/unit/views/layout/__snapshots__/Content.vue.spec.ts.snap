// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders snapshot 1`] = `
<main
  class="w-full"
>
  <transition-stub
    mode="out-in"
    name="fade"
  >
    <router-view-stub
      name="default"
    />
  </transition-stub>
</main>
`;
